instrumentation/opentelemetry-instrumentation-celery/src/opentelemetry/instrumentation/celery/__init__.py

```diff
@@ -183,10 +183,17 @@ def _trace_before_publish(self, *args, **kwargs):
         task = utils.retrieve_task_from_sender(kwargs)
         task_id = utils.retrieve_task_id_from_message(kwargs)
 
-        if task is None or task_id is None:
+        if task_id is None:
             return
 
-        operation_name = f"{_TASK_APPLY_ASYNC}/{task.name}"
+        if task is None:
```

Comment:
A line below attaches the span to the task, which is retrieved later; what happens to it now?

Reply:
You mean that [here](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1407/files#diff-ed0dafdf74eda90912242a368cd05845c26ce3178f34c4fea9816f3ff1e49307R211) `task` could be `None` with these changes?

Reply:
`task` isn't used later, all instances of `task` in this function only referenced `task.name`, and that has been replaced with `task_name`.